pkg/geo/geomfn/force_layout.go

```diff
@@ -17,29 +17,60 @@ import (
 )
 
 // ForceLayout forces a geometry into the given layout.
-// If dimensions are added, 0 coordinates are padded to them.
+// If a Z dimension is added, 0 is added as the Z value.
+// If an M dimension is added, 0 is added as the M value.
 func ForceLayout(g geo.Geometry, layout geom.Layout) (geo.Geometry, error) {
+	return ForceLayoutWithDefaultZM(g, layout, 0, 0)
+}
+
+// ForceLayoutWithDefaultZ forces a geometry into the given layout.
+// If a Z dimension is added, defaultZ is added as the Z value.
+// If an M dimension is added, 0 is added as the M value.
+func ForceLayoutWithDefaultZ(
+	g geo.Geometry, layout geom.Layout, defaultZ float64,
+) (geo.Geometry, error) {
+	return ForceLayoutWithDefaultZM(g, layout, defaultZ, 0)
+}
+
+// ForceLayoutWithDefaultM forces a geometry into the given layout.
+// If a Z dimension is added, 0 is added as the Z value.
+// If an M dimension is added, defaultM is added as the M value.
+func ForceLayoutWithDefaultM(
+	g geo.Geometry, layout geom.Layout, defaultM float64,
+) (geo.Geometry, error) {
+	return ForceLayoutWithDefaultZM(g, layout, 0, defaultM)
+}
+
+// ForceLayoutWithDefaultZM forces a geometry into the given layout.
+// If a Z dimension is added, defaultZ is added as the Z value.
+// If an M dimension is added, defaultM is added as the M value.
+func ForceLayoutWithDefaultZM(
+	g geo.Geometry, layout geom.Layout, defaultZ float64, defaultM float64,
+) (geo.Geometry, error) {
 	geomT, err := g.AsGeomT()
 	if err != nil {
 		return geo.Geometry{}, err
 	}
-	retGeomT, err := forceLayout(geomT, layout)
+	retGeomT, err := forceLayout(geomT, layout, defaultZ, defaultM)
 	if err != nil {
 		return geo.Geometry{}, err
 	}
 	return geo.MakeGeometryFromGeomT(retGeomT)
 }
 
-// forceLayout forces a geom.T into the given layout.
-func forceLayout(t geom.T, layout geom.Layout) (geom.T, error) {
+// forceLayout forces a geom.T into the given layout and uses the specified default Z and M values as needed.
```

Comment:
small nit: we wrap comments to 80 characters (and code to 100 characters). i use the wrap to column plugin for comments https://plugins.jetbrains.com/plugin/7234-wrap-to-column

Reply:
thanks for the tip :) I actually have this plugin installed already, not sure why it didn't wrap the comment automatically. i've manually broken the comment up into two lines.